File: golang/1.array/class_photos.go
package main

import "sort"

func ClassPhotos(b, r []int) bool {
    canTakePhoto := true
    sort.Ints(b)
    sort.Ints(r)
    LI := len(r) - 1;

    br := 'b' 

    if r[LI] > b[LI] { 
        br = 'r' 
    }


    for i:=LI ; i > -1 ; i-- { 
        mr := r[i] 
        mb := b[i]


        if br == 'b' && mr >= mb { 
            canTakePhoto = false ;
            break; 
        } else if br == 'r' && mb >= mr{ 
            canTakePhoto = false ;
            break; 
        }
    }

    return canTakePhoto; 
}

File: ts/1.array/class-photos.ts
// o(nlogn);
export function classPhotos(r: number[], b: number[]) {
    let canTakePhoto = true;
    r.sort((a:number,b:number) => a > b ? 1 : -1); 
    b.sort((a:number,b:number) => a > b ? 1 : -1); 

    let br: 'b' | 'r' ; 

    if (r[r.length - 1] > b[b.length -1])  { 
        br = 'r' ; 
    }else { 
        br = 'b' ; 
    }


    for (let i= r.length - 1 ; i>=0 ; i--) { 
        const mb = b[i] ; 
        const mr = r[i] ; 

        if (br == 'r' && mr <= mb) { 
            canTakePhoto = false ;
            break ;
        }else if (br == 'b' && mr >= mb){ 
            canTakePhoto = false ;
            break ;
        }
    }


    return canTakePhoto ;
}
